qa/tasks/cephfs/test_quiesce.py

```diff
@@ -0,0 +1,727 @@
+
+    def test_quiesce_path_release(self):
+        """
+        That killing the quiesce op properly releases the subvolume so that
+        client IO proceeds.
+        """
+
+        self._configure_subvolume()
+        self._client_background_workload()
+
+        P = self.fs.rank_tell(["ops"])
+        log.debug(f"{P}")
+
+        J = self.fs.rank_tell(["quiesce", "path", self.subvolume])
+        reqid = self.reqid_tostr(J['op']['reqid'])
+        self._wait_for_quiesce_complete(reqid)
+
+        P = self.fs.rank_tell(["ops"])
+        log.debug(f"{P}")
+
+        self.fs.kill_op(reqid)
+
+        P = self.fs.rank_tell(["perf", "dump"])
+        log.debug(f"{P}")
+        requests = P['mds']['request']
+        replies = P['mds']['reply']
+        grants = P['mds']['ceph_cap_op_grant']
+
+        def resumed():
+            P = self.fs.rank_tell(["perf", "dump"])
+            log.debug(f"{P}")
+            try:
+                self.assertLess(requests, P['mds']['request'])
+                self.assertLess(replies, P['mds']['reply'])
+                self.assertLess(grants, P['mds']['ceph_cap_op_grant'])
+                return True
+            except AssertionError:
+                return False
+
+        self.wait_until_true(resumed, 60)
+
+        P = self.fs.rank_tell(["ops"])
+        log.debug(f"{P}")
+
+    def test_quiesce_path_link_terminal(self):
+        """
+        That quiesce on path with an terminal link fails with ENOTDIR even
+        pointing to a valid subvolume.
+        """
+
+        self._configure_subvolume()
+
+        self.mount_a.run_shell_payload("ln -s ../.. subvol_quiesce")
+        path = self.mount_a.cephfs_mntpt + "/subvol_quiesce"
+
+        J = self.fs.rank_tell(["quiesce", "path", path, '--wait'])
+        log.debug(f"{J}")
+        self.assertEqual(J['op']['result'], -20) # ENOTDIR: the link is not a directory
+
+    def test_quiesce_path_link_intermediate(self):
+        """
+        That quiesce on path with an intermediate link fails with ENOTDIR.
+        """
+
+        self._configure_subvolume()
+
+        self.mount_a.run_shell_payload("ln -s ../../.. _nogroup")
+        path = self.mount_a.cephfs_mntpt + "/_nogroup/" + self.QUIESCE_SUBVOLUME
+
+        J = self.fs.rank_tell(["quiesce", "path", path, '--wait'])
+        log.debug(f"{J}")
+        self.assertEqual(J['op']['result'], -20) # ENOTDIR: path_traverse: the intermediate link is not a directory
+
+    def test_quiesce_path_notsubvol(self):
+        """
+        That quiesce on a directory under a subvolume is valid.
+        """
+
+        self._configure_subvolume()
+
+        self.mount_a.run_shell_payload("mkdir dir")
+        path = self.mount_a.cephfs_mntpt + "/dir"
+
+        J = self.fs.rank_tell(["quiesce", "path", path, '--wait'])
+        reqid = self.reqid_tostr(J['op']['reqid'])
+        self._wait_for_quiesce_complete(reqid, path=path)
+        self._verify_quiesce(root=path)
+
+    def test_quiesce_path_regfile(self):
```

Comment:
Maybe we should lift this limitation in the future, WDYT? I don't see any reason for us to prevent quiescing a single file, that should be handled well by the existing code, shouldn't it?

Reply:
Sure, I'll invert the test too.

Reply:
Ack